test/common/services/kibana_server/extend_es_archiver.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const ES_ARCHIVER_LOAD_METHODS = ['load', 'loadIfNeeded'];
+const KIBANA_INDEX = '.kibana';
+
+export function extendEsArchiver({ getService, esArchiver }) {
+  const defaults = getService('config').get('uiSettings.defaults');
```

Comment:
Would it make sense to just pass `defaults` and `uiSettings` as arguments into `extendEsArchiver`? Seems like a bit of a code smell to pass someone a tool-fetching assistant when you could just hand them the tool they want.

Reply:
Yeah, that was my original implementation, but the `kibanaServer` service might not be available, so I'm relying on the fact that the `uiSettings.defaults` setting is used to determine that I should load the `kibanaServer` service. This meant that I had to encode into the `esArchiver` when the `kibanaServer` would be interested in extending it. Rather than basing it on the existence of the `uiSettings.defaults` config I had added a `hasService()` method to the service loader that `esArchiver` was using to check for the `kibanaServer` service before calling it's `extendEsArchiver()` function... But that felt like a lot of changes just to avoid passing the `getService()` function to `extendEsArchiver()`. Happy to go back to it if you think that's a better approach.

Reply:
Welp, thought about it again and it really wasn't that much trouble, implemented in c149e11

Reply:
Oh I see. I think it's OK as-is. Could you leave a comment on line 31 to explain this relationship?

```js
// The kibanaServer service won't be available unless uiSettings.defaults has been set.
const { uiSettings } = getService('kibanaServer');
```